kubernetes_asyncio/config/incluster_config.py

```diff
@@ -84,10 +87,29 @@ def _load_config(self):
     def _set_config(self, configuration):
         configuration.host = self.host
         configuration.ssl_ca_cert = self.ssl_ca_cert
-        configuration.api_key['BearerToken'] = "Bearer " + self.token
+        if self.token is not None:
+            configuration.api_key['BearerToken'] = self.token
+        if not self._try_refresh_token:
+            return
+
+        def load_token_from_file(*args):
+            if self.token_expires_at <= datetime.datetime.now():
+                self._read_token_file()
+            return self.token
+
+        configuration.get_api_key_with_prefix = load_token_from_file
```

Comment:
The Configuration class has an attribute named `refresh_api_key_hook` which can be used to refresh token.